### PR TITLE
storage: Allow truncation of ID column

### LINKED ISSUE #19767: Layout problem with new storage view
### Explain what happens

grafik

### Version of Cockpit

307

### Where is the problem in Cockpit?

Storage

### Server operating system

Debian

### Server operating system version

12.3

### What browsers are you using?

Firefox

### System log

_No response_

### PR DESCRIPTION
**Pull request by mvollmer**
And redo indentation magic. Now also tables without icons will show indentations, and the Truncate element will get a correct width on indented columns.

Thanks a lot to MohdMohsin97 for the initial version of this!

Fixes #19767

### CODE REVIEW

**Comment by mvollmer** [comment]
This was supposed to have a test... I lost it somehow.

File: pkg/storaged/pages.jsx [hunk 1]
```diff
@@ -591,7 +592,7 @@ export const PageTable = ({ emptyCaption, aria_label, pages, crossrefs, sorted,
                     <CardBody>
                         <Split hasGutter>
                             { icon && <SplitItem>{icon}</SplitItem> }
-                            <SplitItem isFilled><strong>{name}</strong>{info}</SplitItem>
+                            <SplitItem isFilled><strong><Truncate content={name} /></strong>{info}</SplitItem>
```
Comment (cockpituous) [inline pkg/storaged/pages.jsx:595]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19919-20240131-124744-6f2c1986-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
Looks like the test changes can not run twice in a row...

File: pkg/storaged/pages.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/pages.jsx:595]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19919-20240202-142745-1d909c55-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-truncate.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.ct-no-truncate-min-width {
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply pkg/lib/cockpit-components-truncate.scss:1]: Alright!

File: pkg/storaged/pages.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/pages.jsx:595]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19919-20240205-135746-9af64d76-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Comment by mvollmer** [comment]
Damn pixel test conflicts, rebased.

**Review by jelly — APPROVED** [review]

**Comment by jelly** [comment]
> Damn pixel test conflicts, rebased.

all green now \o/

File: pkg/storaged/pages.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/pages.jsx:595]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19919-20240207-072057-9f475cc9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**